proto/src/determined/api/v1/api.proto

```diff
@@ -2183,4 +2183,62 @@ service Determined {
       tags: [ "Experiments", "Internal" ]
     };
   }
+
+  // Bind resource pool to workspace
+  rpc BindRPToWorkspace(BindRPToWorkspaceRequest)
+      returns (BindRPToWorkspaceResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/resource-pools/{resource_pool_name}/workspace-bindings",
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+
+  // Unbind resource pool to workspace
+  rpc UnbindRPFromWorkspace(UnbindRPFromWorkspaceRequest)
+      returns (UnbindRPFromWorkspaceResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/resource-pools/{resource_pool_name}/delete-workspace-bindings",
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+
+  // Overwrite resource pool - workspace bindings
+  rpc OverwriteRPWorkspaceBindings(OverwriteRPWorkspaceBindingsRequest)
+      returns (OverwriteRPWorkspaceBindingsResponse) {
+    option (google.api.http) = {
+      put: "/api/v1/resource-pools/{resource_pool_name}/workspace-bindings",
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+
+  // List all resource pools bound to a specific workspace
+  rpc ListRPsBoundToWorkspace(ListRPsBoundToWorkspaceRequest)
+      returns (ListRPsBoundToWorkspaceResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/workspaces/{workspace_id}/available-resource-pools"
```

Comment:
yeah the tricky part about this endpoint is it isn't just describing the bound pools - it is decribing all resource pools that the workspace can access. So that is counting pools bound to the workspace + unbound pools.